packages/core/src/domain/session/sessionManager.ts

```diff
@@ -17,12 +18,15 @@ export interface SessionManager<TrackingType extends string> {
   ) => SessionContext<TrackingType> | undefined
   renewObservable: Observable<void>
   expireObservable: Observable<void>
+  sessionStateUpdateObservable: Observable<{ previousState: SessionState; newState: SessionState }>
   expire: () => void
+  updateSessionState: (state: Partial<SessionState>) => void
 }
 
 export interface SessionContext<TrackingType extends string> extends Context {
   id: string
   trackingType: TrackingType
+  isReplayForced: boolean
```

Comment:
**💭 thought:** ‏Ideally we should remove any notion of "replay" from `@datadog/browser-core`. But I understand this last bit is not trivial to fix, so let's keep it like this for now, and we'll re-explore this later (I opened https://datadoghq.atlassian.net/browse/RUM-4740 for that)